RegistryManager.SerializeCurrentInstall changes

RichardLake opened the pull request:
Change depends to recommends.
  Allows user customization of the install in the GUI
Do not output provided versions.
  They are no real mods.

Closes KSP-CKAN/CKAN#912.
Relevant to KSP-CKAN/CKAN#972.

RichardLake committed:
RegistryManager.SerializeCurrentInstall changes

Change depends to recommends.
  Allows user customization of the install in the GUI
Do not output provided versions.
  They are no real mods.

pjf commented:
This looks perfect for KSP-CKAN/CKAN#912.

Regarding the broader scope, there's a real question as to what these files are used for. Are they:

* A way to duplicate an install on another machine? In which case we *want* depends.
* A way to share favourites, in which case we want a softer recommends?

As per KSP-CKAN/CKAN#972, I imagine they're being used for both. In this case, it would be lovely to have code take an option indicating which type of file we should generate. The default I imagine would be hard depends with version numbers, but with an option of instead generating recommends without version numbers. (Or bonus points, have options for depends/recommends, and versions/no-versions separately).

RichardLake committed:
Create CurrentInstallAsCKAN method.

Will allow clients to avoid having to use installed-default.ckan and has greater flexibility.

RichardLake commented:
Made changes as per suggestions.
